### PR TITLE
Document data retention/data privacy

### PR DESCRIPTION
**Pull request by benolayinka**
#### Summary

Document data retention re https://github.com/TheThingsIndustries/lorawan-stack-support/issues/463

#### Notes for Reviewers

Please let me know if you have any unique insight in relation to data privacy that should be included here

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- x ] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [x] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/reference/data-retention/_index.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: "Data Retention and Privacy"
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:2]: This is strictly about TTSC, right? Should we clarify that?

File: doc/content/reference/data-retention/_index.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+{{% tts %}} Application Server stores
+
+- Recent uplink application payloads, in a Redis database
+
+{{% tts %}} Cloud runs on an isolated subnet, and the databases can only be accessed from within the instance.
+
+{{% tts %}} uses your Network Key to decrypt LoRaWAN packets - these can only be decrypted at the Network Server, not by gateways or anywhere else along the IP route they travel.
+
+{{% tts %}} uses your Application Key if provided to decrypt the application payload, which is available in the [JSON message]({{< ref "reference/data-formats" >}}). It is also possible to skip payload decryption in {{% tts %}}, and perform decryption later in your application pipeline. Select **Skip payload encryption/decryption** when registering a device to do so.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:23]:
```suggestion
{{% tts %}} uses the LoRaWAN AppSKey to decrypt the application payload, which is available in the [JSON message]({{< ref "reference/data-formats" >}}). It is also possible to skip payload decryption in {{% tts %}}, and perform decryption later in your application pipeline. Select **Skip payload encryption/decryption** when registering a device to do so.
```

File: doc/content/reference/data-retention/_index.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+{{< figure src="skip-encryption.png" >}}
+
+The Things Industries will never read application payloads, even if you provide your Application Key.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:27]:
```suggestion
The Things Industries will never read application payloads, even if the AppSKey is configured.
```

Our services do process unencrypted payload. What do you mean with "The Things Industries", the people? Support and operations will sometimes need to access data streams on the application level.

Comment (benolayinka) [reply doc/content/reference/data-retention/_index.md:27]: https://github.com/TheThingsIndustries/lorawan-stack-support/issues/402#issuecomment-825763564

Comment (benolayinka) [reply doc/content/reference/data-retention/_index.md:27]: should we say "The Things Industries personnel will never access your application payloads, unless given permission to do so to provide support."

**Review by htdvisser — COMMENTED** [review]
I think this is confusing and that we need to create a better separation between The Things Stack (what it stores, what admins can access, and how long application data / events / ... are retained) and deployments such as The Things Stack Community / (Dedicated) Cloud (or even deployments of partners / other operators).

I think we should also make clear that tenants in multi-tenant deployments are completely isolated, so admins in another tenant don't have access to data outside their tenant's scope.

I think we need to make clear that different operators can have different configuration for admin access and data retention. In the future this may even be configurable per tenant.

I think we also need to explain what is / isn't shared when peering through Packet Broker is enabled.

**Comment by adriansmares** [comment]
To add on @htdvisser's comment: Maybe it would be a good time to define the stake holders of a stack deployment, such that we can refer back to them while dealing with such questions. What I mean by this is clarifying who are the following:
- Infrastructure owners and maintainers. For TTSC this is The Things Industries, and formally we do have access to everything (as database admins), albeit this kind of access is not used in practice. For TTN this is The Things Network, albeit in practice still TTI. They do have access to all of the tenants in a deployment.
- Network administrators (users with `admin` access) in a tenant. They can access all of the entities in the tenant, but nothing outside the tenant. They have access to end device secrets too.
- Users and Organizations in a tenant. They can access entities for which they are collaborators, and their access is limited by their permissions.

This is a bit of scope creep, but I think it is very relevant for such queries - in my experience users do not really understand (nor is it communicated to them) what is the difference between being a stack user with the admin checkbox filled in and actually the kind of access that an infrastructure owner has.

**Comment by benolayinka** [comment]
> I think this is confusing and that we need to create a better separation between The Things Stack (what it stores, what admins can access, and how long application data / events / ... are retained) and deployments such as The Things Stack Community / (Dedicated) Cloud (or even deployments of partners / other operators).
> 
> I think we should also make clear that tenants in multi-tenant deployments are completely isolated, so admins in another tenant don't have access to data outside their tenant's scope.
> 
> I think we need to make clear that different operators can have different configuration for admin access and data retention. In the future this may even be configurable per tenant.
> 
> I think we also need to explain what is / isn't shared when peering through Packet Broker is enabled.

I tried to capture as much of this as possible. It would be good to add some clues about what kind of data retention policies are configurable for customer deployed stacks.

For packet broker, do you mean we should list what information will be persisted if the user chooses to share it?

File: doc/content/reference/data-retention/_index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+This reference contains information about what access administrators and users have to data, and how long {{% tts %}} persists device data. For inquiries about data privacy or data security, read on, or [contact The Things Industries](cloud@thethingsindustries.com).
+
+<!--more-->
+
+## Important Terms
+
+To understand how data is secured in {{% tts %}} it is necessary to understand some important roles:
+
+**Infrastructure Owners**: Those with access to the hardware {{% tts %}} runs on. For {{% tts %}} Cloud and Community Edition, this is The Things Industries. For Enterprise and Open Source deployments, this is you, the customer. Formally, the **Infrastructure Owner** has access to all data stored in a database on the hardware. In practice, The Things Industries does not use this to access data in {{% tts %}} Cloud or Community Edition deployments.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:15]:
```suggestion
**Infrastructure Owners**: Those with access to the hardware {{% tts %}} runs on. For {{% tts %}} Cloud, Dedicated Cloud and Community Edition, this is The Things Industries. {{% tts %}} Enterprise and Open Source deployments are managed by the customer and user respectively; The Things Industries does not have any infrastructural access. Technically, **Infrastructure Owners** have access to all data stored in databases or persistent volumes. In practice, The Things Industries uses this direct data access to {{% tts %}} Cloud or Community Edition deployments only for backups.
```

File: doc/content/reference/data-retention/_index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+**Network Administrators**: Users with `admin` access in a tenant. **Network Administrators** have access to all entities in a tenant, including device secrets. In multi-tenant environments, tenants are completely isolated, and **Network Administrators** of one tenant have **no access** to other tenants.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:17]:
```suggestion
**Network Administrators**: Users with administrative access in a tenant. **Network Administrators** have access to all entities in a tenant, including device secrets. In multi-tenant environments, tenants are completely isolated, and **Network Administrators** of one tenant have **no access** to other tenants.
```

File: doc/content/reference/data-retention/_index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+**Users and Organizations**: Users without `admin` access, and organizations, can access entities which they have [collaborator rights]({{< ref "getting-started/user-management" >}}) over.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:19]:
```suggestion
**Users and Organizations**: Users without administrative access, and organizations, can access entities which they have [collaborator rights]({{< ref "getting-started/user-management" >}}) over.
```

File: doc/content/reference/data-retention/_index.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:20]: Add section support engineers? I think it's good to mention that TTI support engineers do not have infrastructural access, and customers should allow them explicitly and temporarily to access applications and gateways in their tenant.

File: doc/content/reference/data-retention/_index.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+## Cloud Deployments
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:21]:
```suggestion
## {{% tts %}} Cloud and Dedicated Cloud
```

File: doc/content/reference/data-retention/_index.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+The following information applies to {{% tts %}} deployments hosted by The Things Industries, specifically {{% tts %}} Cloud and Dedicated Cloud.
+
+{{% tts %}} deployments hosted on customer hardware, specifically {{% tts %}} Enterprise, are completely managed by the customer and can be configured to meet more specific data privacy requirements.
+
+## Persisted Data
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:27]:
```suggestion
### Persisted Data
```

File: doc/content/reference/data-retention/_index.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+On {{% tts %}} Cloud and Dedicated Cloud, the Network Server stores
+
+- Recent uplinks as part of the MAC state, in a Redis database
+
+The Application Server stores
+
+- Recent uplink application payloads, in a Redis database
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:35]: Is this really necessary to mention?

If so, we should also mention that Application Server stores upstream messages through the Storage Integration (+ link) if that is enabled. Also mention the data retention here.

File: doc/content/reference/data-retention/_index.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+{{% tts %}} Cloud runs on an isolated subnet, and the databases can only be accessed from within the instance.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:37]:
```suggestion
{{% tts %}} Cloud runs in isolated subnets. Databases can only be accessed from within the subnet and are not publicly accessible.
```

File: doc/content/reference/data-retention/_index.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+{{% tts %}} uses your Network Key to decrypt LoRaWAN packets - these can only be decrypted at the Network Server, not by gateways or anywhere else along the IP route they travel.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:39]:
```suggestion
{{% tts %}} uses your NwkSKey (NwkSEncKey in LoRaWAN 1.1) to decrypt LoRaWAN network-level payload. These session keys are only known to the Network Server and are never exposed downstream (to gateways) or upstream (to applications).
```

File: doc/content/reference/data-retention/_index.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+{{% tts %}} uses the LoRaWAN AppSKey to decrypt the application payload, which is available in the [JSON message]({{< ref "reference/data-formats" >}}). It is also possible to skip payload decryption in {{% tts %}}, and perform decryption later in your application pipeline. Select **Skip payload encryption/decryption** when registering a device to do so.
+
+{{< figure src="skip-encryption.png" >}}
+
+The Things Industries personnel will never access your application payloads, even if the AppSKey is configured, unless given permission to do so to provide support.
```
Comment (johanstokking) [inline doc/content/reference/data-retention/_index.md:45]: I think this line can be replaced by mentioning above what support engineers can and cannot do

**Review by adriansmares — APPROVED** [review]
Nothing to add. Please address the comments from @johanstokking .

**Comment by benolayinka** [comment]
added @johanstokking suggestions. do we need any information about packet broker?

**Comment by johanstokking** [comment]
> added @johanstokking suggestions. do we need any information about packet broker?

Not here I think.

**Review by johanstokking — APPROVED** [review]

**Pull request merged**